File: oss/src/main/java/io/tn/oss/amazon/S3Args.java
package io.tn.oss.amazon;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.google.gson.annotations.SerializedName;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class S3Args {
    @JsonProperty("Version")
    @SerializedName("Version")
    String version;
    @JsonProperty("Statement")
    @SerializedName("Statement")
    List<S3StatementArgs> statement = new ArrayList<>();
}

File: oss/common/src/main/kotlin/net/yan100/compose/oss/common/S3Args.kt
package net.yan100.compose.oss.common

import com.fasterxml.jackson.annotation.JsonProperty

data class S3Args(
  @JsonProperty("Version")
  var version: String? = null,
  @JsonProperty("Statement")
  var statement: MutableList<S3StatementArgs> = ArrayList(),
)
